geonode/metadata/handlers/abstract.py

```diff
@@ -0,0 +1,149 @@
+        in case the handler needs some cross related data contained in the resource.
+        """
+        pass
+
+    def load_serialization_context(self, resource: ResourceBase, jsonschema: dict, context: dict):
+        """
+        Called before calls to get_jsonschema_instance in order to initialize info needed by the handler
+        """
+        pass
+
+    def load_deserialization_context(self, resource: ResourceBase, jsonschema: dict, context: dict):
+        """
+        Called before calls to update_resource in order to initialize info needed by the handler
+        """
+        pass
+
+    def _add_subschema(self, jsonschema, property_name, subschema, after_what=None):
+        after_what = after_what or subschema.get("geonode:after", None)
+
+        if not after_what:
+            jsonschema["properties"][property_name] = subschema
+        else:
+            ret_properties = {}
+            added = False
+            for key, val in jsonschema["properties"].items():
+                ret_properties[key] = val
+                if key == after_what:
+                    ret_properties[property_name] = subschema
+                    added = True
+
+            if not added:
+                logger.warning(f'Could not add "{property_name}" after "{after_what}"')
+                ret_properties[property_name] = subschema
+
+            jsonschema["properties"] = ret_properties
+
+    @deprecated("Use _add_subschema instead")
+    def _add_after(self, jsonschema, after_what, property_name, subschema):
+        ret_properties = {}
+        added = False
+        for key, val in jsonschema["properties"].items():
+            ret_properties[key] = val
+            if key == after_what:
+                ret_properties[property_name] = subschema
+                added = True
+
+        if not added:
+            logger.warning(f'Could not add "{property_name}" after "{after_what}"')
+            ret_properties[property_name] = subschema
+
+        jsonschema["properties"] = ret_properties
+
+    @staticmethod
+    def _set_error(errors: dict, path: list, msg: str):
+        logger.info(f"Reported message: {'/'.join(path)}: {msg} ")
+        elem = errors
+        for step in path:
+            elem = elem.setdefault(step, {})
+        elem = elem.setdefault("__errors", [])
+        elem.append(msg)
+
+    @staticmethod
+    def _localize_label(context, lang: str, text: str):
+        # Try localization via thesaurus:
+        label = context["labels"].get(text, None)
+        # fallback: gettext()
+        if not label:
+            label = _(text)
+
+        return label
+
+    @staticmethod
+    def _localize_subschema_label(context, subschema: dict, lang: str, annotation_name: str):
+        if annotation_name in subschema:
+            subschema[annotation_name] = MetadataHandler._localize_label(context, lang, subschema[annotation_name])
+
+
+class MetadataFieldException(Exception):
```

Comment:
I guess to make it more linear, will be better to move this 3 exception is another file `exceptions.py`